src/Track.h

```diff
@@ -367,17 +370,27 @@ class AUDACITY_DLL_API Track /* not final */
 
    std::shared_ptr<TrackList> GetOwner() const { return mList.lock(); }
 
+   LinkType GetLinkType() const noexcept;
+   //! Returns true if next track is linked to this one
+   bool HasLinkedTrack() const noexcept;
+   //! Returns true if the track is aligned with previous one
+   bool IsAlignedWithLeader() const;
```

Comment:
Replace "previous one" with "its leader" and this comment is not objectionable.

But I still strongly disagree with the unnecessary disturbance of the places that called GroupChannels().

Reply:
>You may have a justification. What is it?

I was studying what GroupChannel does for multi-channel tracks and came to conclusion that it does almost nothing: beeing called for multi-channel tracks it'll just set channel type to mono (as it was before) and will set the link to false (as it was before). The result of GroupChannels is a set of ungrouped channels, which is a bit misleading i think, and will lead to a false conclusions about the behaviour in future (we don't know yet how 5.1 channels would be grouped, for example). Maybe that's another reason to have a StereoWaveTrack type, then grouping and checks for leads will become redundant? But the consequences would have their own downsides surely

Reply:
I still disapprove what you did.  You must consider the implementation and also the interface.

There is what GroupChannel does now, which does not yet implement multi-channel tracks, because we don't yet have a data representation for them.

However there is all the other code using the abstractions provided by Track and TrackList, that ARE programmed to an interface anicipating possible multi-channel implementation in the future.  It required a lot of my work in previous years to make that so -- that the assumption of at-most-stereo tracks is made in very few places.

That should remain so.  Please change the interface of GroupChannel only so you specify alignment or not.  Changing that function into two functions that can only allow for making either mono and stereo tracks is not a change in behavior now, but it is not an improvement of the interface.

Reply:
I do agree that you can simplify away the resetChannels argument which was never non-default.

Reply:
>However there is all the other code using the abstractions provided by Track and TrackList, that ARE programmed to an interface anticipating possible multi-channel implementation in the future.

We will need to make changes to GroupChannels then (maybe multiple times), and this will affect each piece of the code which calls this method, and make sure that we don't break anything unintentionally: to add a behaviour which will handle grouping for more than two channels, we'll need to interfere the code that does grouping for two channels.